### PR TITLE
Allow check for fingerprint anywhere in the certificate signing chain

### PR DESCRIPTION
**Pull request by cvillerm**
I added a new API "checkInCertChain" allowing fingerprints to be checked against the server certificate or any certificate in the signing chain.
This allow pinning to a Certificate Authority certificate what can be good enough from a security viewpoint and would avoid changing fingerprints in case of server certificate renewal is if same CA is used.

### CODE REVIEW

**Comment by EddyVerbruggen** [comment]
Sorry this will break backward compatibility. If you add a new function like `checkInChain` next to `check` then we don't break things.

**Comment by cvillerm** [comment]
I indeed added a new function checkInCertChain in the JS code. It's only the native function that I shared between the 2 JS call. Isn't this Ok?

**Comment by EddyVerbruggen** [comment]
You are so right, I only checked your PR with a half eye. Sorry.

File: src/ios/SSLCertificateChecker.m [hunk 1]
```diff
@@ -91,14 +116,14 @@ @implementation SSLCertificateChecker
...
-                             }
-                             }
-                             
-                             @end
+    
+    CustomURLConnectionDelegate *delegate =
+    [[CustomURLConnectionDelegate alloc] initWithPlugin:self callbackId:command.callbackId checkInCertChain:[[command.arguments objectAtIndex:1] boolValue] allowedFingerprint:[command.arguments objectAtIndex:2] allowedFingerprintAlt:[command.arguments objectAtIndex:3]];
+    
+    if (![NSURLConnection connectionWithRequest:request delegate:delegate]) {
+ns        CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_JSON_EXCEPTION messageAsString:@"CONNECTION_FAILED"];
```
Comment (EddyVerbruggen) [inline src/ios/SSLCertificateChecker.m]: Compile error, I will fix it

File: src/ios/SSLCertificateChecker.m [hunk 2]
```diff
@@ -7,44 +7,67 @@ @interface CustomURLConnectionDelegate : NSObject <NSURLConnectionDelegate>;
...
+    
+    CFIndex count = 1;
+    if (self._checkInCertChain) {
+        count = SecTrustGetCertificateCount(trustRef);
+    }
+    
+    for (CFIndex i = 0; i < count; i++)
+    {
+        SecCertificateRef certRef = SecTrustGetCertificateAtIndex(trustRef, i);
+        CFStringRef certSummary = SecCertificateCopySubjectSummary(certRef);
```
Comment (EddyVerbruggen) [inline src/ios/SSLCertificateChecker.m]: I'm guessing this line can be removed

**Pull request merged**